pkg/volume/empty_dir/empty_dir_linux.go

```diff
@@ -35,18 +36,33 @@ type realMountDetector struct {
 	mounter mount.Interface
 }
 
-func (m *realMountDetector) GetMountMedium(path string) (storageMedium, bool, error) {
-	glog.V(5).Infof("Determining mount medium of %v", path)
-	notMnt, err := m.mounter.IsLikelyNotMountPoint(path)
+func (m *realMountDetector) GetMountMedium(p string) (storageMedium, bool, error) {
+	glog.V(5).Infof("Determining mount medium of %v", p)
+	notMnt, err := m.mounter.IsLikelyNotMountPoint(p)
 	if err != nil {
-		return 0, false, fmt.Errorf("IsLikelyNotMountPoint(%q): %v", path, err)
+		return 0, false, fmt.Errorf("IsLikelyNotMountPoint(%q): %v", p, err)
 	}
+
+	// Workaround buggy filesystems like vmhgfs which fail on statfs with long
+	// file names, walk up the tree stopping just in front of the filesystem border
+	stat := syscall.Stat_t{}
+	if err = syscall.Stat(p, &stat); err != nil {
+		return 0, false, fmt.Errorf("stat(%q): %v", p, err)
+	}
+	dev := stat.Dev
+	dotdot := p
+	for dotdot != "/" && err == nil && dev == stat.Dev {
+		p, dotdot = dotdot, path.Dir(p)
+		dev = stat.Dev
+		err = syscall.Stat(dotdot, &stat)
+	}
+
 	buf := syscall.Statfs_t{}
-	if err := syscall.Statfs(path, &buf); err != nil {
-		return 0, false, fmt.Errorf("statfs(%q): %v", path, err)
+	if err := syscall.Statfs(p, &buf); err != nil {
```

Comment:
Is this acknowledged as a bug (to be fixed) or correct behavior
(delusional?)

On Mon, Oct 19, 2015 at 3:10 PM, Dr. Stefan Schimanski <
notifications@github.com> wrote:

> In pkg/volume/empty_dir/empty_dir_linux.go
> <https://github.com/kubernetes/kubernetes/pull/15849#discussion_r42432641>
> :
>
> >  	buf := syscall.Statfs_t{}
> > -	if err := syscall.Statfs(path, &buf); err != nil {
> > -		return 0, false, fmt.Errorf("statfs(%q): %v", path, err)
> > +	if err := syscall.Statfs(p, &buf); err != nil {
>
> 200 characters. So with some uuid in the path this can be easily reached.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/15849/files#r42432641>.
>